File: app/src/main/java/com/sstechcanada/todo/custom_views/MasterIconGridItemView.java
package com.sstechcanada.todo.custom_views;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.widget.FrameLayout;
import android.widget.ImageView;

import androidx.cardview.widget.CardView;

import com.sstechcanada.todo.R;

public class MasterIconGridItemView extends FrameLayout {

    private ImageView iconImageView;
    private CardView cardView;

    public MasterIconGridItemView(Context context) {
        super(context);
        LayoutInflater.from(context).inflate(R.layout.item_grid_master_list, this);
        iconImageView = (ImageView) getRootView().findViewById(R.id.iconImageView);
        cardView = (CardView) getRootView().findViewById(R.id.cardView);
    }

    public void display(int text,boolean isSelected) {
//        int imageResource = getContext().getResources().getIdentifier(text, null, getContext().getPackageName());
        Drawable drawable = getContext().getResources().getDrawable(text);
        iconImageView.setBackground(drawable);
//        textView.setText(text);
        display(isSelected);
    }

    public void display(boolean isSelected) {
        cardView.setBackgroundResource(isSelected ? R.drawable.selected_icon_bg : R.color.colorUncompletedBackground);
    }
}

File: app/src/main/java/com/sstechcanada/todo/custom_views/MasterIconGridItemView.kt
package com.sstechcanada.todo.custom_views

import android.content.Context
import android.widget.FrameLayout
import androidx.cardview.widget.CardView
import com.sstechcanada.todo.R
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView

class MasterIconGridItemView(context: Context?) : FrameLayout(
    context!!
) {
    private val iconImageView: ImageView
    private val cardView: CardView
    fun display(text: Int, isSelected: Boolean) {
//        int imageResource = getContext().getResources().getIdentifier(text, null, getContext().getPackageName());
        val drawable = context.resources.getDrawable(text)
        iconImageView.background = drawable
        //        textView.setText(text);
        display(isSelected)
    }

    fun display(isSelected: Boolean) {
        cardView.setBackgroundResource(if (isSelected) R.drawable.selected_icon_bg else R.color.colorUncompletedBackground)
    }

    init {
        LayoutInflater.from(context).inflate(R.layout.item_grid_master_list, this)
        iconImageView = rootView.findViewById<View>(R.id.iconImageView) as ImageView
        cardView = rootView.findViewById<View>(R.id.cardView) as CardView
    }
}